src/openrct2/rct1/S4Importer.cpp

```diff
@@ -1243,6 +1243,15 @@ class S4Importer final : public IParkImporter
             }
         }
 
+        VEHICLE_STATUS statusSrc = VEHICLE_STATUS_MOVING_TO_END_OF_STATION;
+
+        // This will need to be updated if we ever add more VEHICLE_STATUS enums.
+        if (src->status >= 0 || src->status <= 30)
```

Comment:
The problem is that src->status is unsigned. Just drop the `src->status>=0` from this comparison. This line should be:
```
if (src->status <= static_cast<uint8_t>(VEHICLE_STATUS_STOPPED_BY_BLOCK_BRAKES))
```
Ideally there would be a way to get the size of the enum but that is not currently something in the standard. When we switch it to an enum class we will add a constexpr for enum class size.

Reply:
That was the first thing I tried, but it failed too. It's possible that I misinterpreted the errors that time, so I'll try it again.